app/controllers/api/v1/accounts_controller.rb

```diff
@@ -1,7 +1,8 @@
 # frozen_string_literal: true
 
 class Api::V1::AccountsController < ApiController
-  before_action -> { doorkeeper_authorize! :read }, except: [:follow, :unfollow, :block, :unblock, :mute, :unmute, :update_credentials]
+  before_action -> { doorkeeper_authorize_if_got_token! :read }, only: [:show, :following, :followers, :statuses, :search]
+  before_action -> { doorkeeper_authorize! :read }, except: [:show, :following, :followers, :statuses, :search, :follow, :unfollow, :block, :unblock, :mute, :unmute, :update_credentials]
```

Comment:
I agree that ```:only``` would look nicer.
I have used ```:except``` for the case when a new action is added and whoever does this forgets to add this new action to one of ```doorkeeper_authorize``` lines, so it falls automatically in the one with ```:except``` and at least some authorization is done.